.github/workflows/pr.yml

```diff
@@ -0,0 +1,52 @@
+name: Pull Request
+
+on:
+  workflow_dispatch:
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [18.x, 20.x]
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Update NPM Version
+        run: npm i -g npm
+
+      - name: Cache dependencies
+        uses: actions/cache@v4
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ matrix.node-version }}-${{ hashFiles('**/package-lock.json') }}
```

Comment:
I'm not a fan of using package-lock on library repos, there is value in catching regressions in compatibility with third party libraries

Reply:
I tend to always commit to make sure which version of dependencies are being used, and to be easier to cache node modules on CI.

But since I use pinned version of the packages, I don't think there is a difference of using or not except from the cache perspective.

Reply:
I tend to always commit to make sure which version of dependencies are being used, and to be easier to cache node modules on CI.

But since I use pinned version of the packages, I don't think there is a difference of using or not except from the cache perspective.

Reply:
caching is not very beneficial for small repos, CI is going to be fast either way

Reply:
Ok, once I got home I can remove those caching.

If you have time, you can merge as it is and just create two commits to fix both comments.